spec/features/ticket_purchases_spec.rb

```diff
@@ -108,6 +109,49 @@ def make_failed_stripe_purchase
         expect(purchase.paid).to be true
       end
 
+      scenario 'purchases a free registartion ticket' do
+        visit root_path
+        click_link 'Register'
+
+        expect(current_path).to eq(new_conference_conference_registration_path(conference.short_title))
+        click_button 'Register'
+
+        fill_in "tickets__#{first_registration_ticket.id}", with: '1'
+        expect(current_path).to eq(conference_tickets_path(conference.short_title))
+
+        click_button 'Continue'
+        expect(current_path).to eq(new_conference_conference_registration_path(conference.short_title))
+        expect(flash).to eq('Thanks! Your ticket is booked successfully. Please register for the conference.')
+
+        purchase = TicketPurchase.where(user_id: participant.id, ticket_id: first_registration_ticket.id).first
+        expect(purchase.quantity).to eq(1)
+        expect(purchase.paid).to be true
+      end
+
+      scenario 'purchases a non-free registartion ticket' do
```

Comment:
these tests are pretty long and it seems like each scenario is testing for multiple things. Maybe you can separate groups of expect statements into their own tests so you test suite can be more descriptive and might help with how things calculate test coverage (not sure about that one though).